docs/src/main/asciidoc/includes/tracing/common-callbacks.adoc

```diff
@@ -0,0 +1,159 @@
+///////////////////////////////////////////////////////////////////////////////
+
+    Copyright (c) 2024 Oracle and/or its affiliates.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+ifndef::rootdir[:rootdir: {docdir}/../..]
+include::{rootdir}/attributes.adoc
+
+// tag::defs[]
+:tracing-javadoc: {tracing-javadoc-base-url}/io/helidon/tracing
+:tracing-api-javadoc: {tracing-javadoc}
+:listener-type: SpanLifeCycleListener
+// end::defs[]
+
+
+// tag::detailed[]
+// tag::intro[]
+== Responding to Span Life Cycle Events [[Tracing-callbacks]]
+
+Applications and libraries can register listeners to be notified at several moments during the lifecycle of every span:
+
+* Before a new span starts
```

Comment:
My hackles go up here: every other event is an "after". What makes this special, I wonder?

Reply:
Because it is passed the `Span.Builder` `beforeStart` can invoke `Span.Builder#unwrap` (which might be ill-advised in general but there it is) to do something interesting with the underlying implementation's span builder before the span is started, whether that's retrieving information about the builder or setting something in the builder that affects the soon-to-be-created span.

One could argue we could also expose `beforeActivate(Span)` in addition to `afterActivate(Span, Scope)` for a similar reason. I'd tend to resist that just because it turns out there's not too much interesting in activating a span that a developer's code could usefully affect. But it does result in `beforeStart` standing out compared to the others.

Reply:
I think I see; so in old school apps you'd have a `SpanBuilderListener` as well, and this method would belong to it so that the `SpanListener` would only deal with `Span`s and not their associated builders, but here that was deemed (not unreasonably) to be too many classes etc., I presume?

Reply:
To me simplicity argued for one listener type which Helidon would inform about all aspects of each span's life cycle, from creation (which just happens to be via a `Span.Builder`) through end. Adding a separate listener type to listen to the one `Span.Builder` event of interest seems unnecessary. 

Maybe `TracingListener` would be an alternative name to more clearly communicate the broader set of events the listener reacts to--not just span events but span builder ones as well.